lib/validators/movie.js

```diff
@@ -2,7 +2,15 @@
 
 const Joi = require('joi');
 
-module.exports = Joi.object().keys({
+exports.payload = Joi.object().keys({
   title: Joi.string().min(1).max(255).required(),
   release_year: Joi.number().integer().min(1878).max(9999).optional()
 });
+
+exports.query = Joi.object().keys({
+  title: Joi.string().min(1).max(255).optional(),
```

Comment:
Ideally, I don't think you'd want to be able to search by both `title` and `fuzzy_title` but only by one of them. We can enforce this constraint using Joi's [`.xor()`](https://github.com/hapijs/joi/blob/master/API.md#objectxorpeers) property. This will enforce that only one of the keys (or neither) is present at any given time.

Example:
```js
exports.query = Joi.object().keys({
  title: Joi.string().min(1).max(255).optional(),
  fuzzy_title: Joi.string().min(1).max(255).optional(),
  // other keys here ...
})
.xor('title', 'fuzzy_title');
```